noir-projects/aztec-nr/uint-note/src/uint_note.nr

```diff
@@ -29,4 +100,223 @@ impl UintNote {
+    /// possible to verify the integrity of its contents due to it hiding information. The recommended ways to do this
+    /// are to retrieve it from public storage, or to receive it in an internal public function call.
+    ///
+    /// Each partial note should only be used once, since otherwise multiple notes would be linked together and known to
+    /// belong to the same owner.
+    ///
+    /// As part of the partial note cration process, a log will be sent to `recipient` from `sender` so that they can
+    /// discover the note. `recipient` will typically be the same as `owner`.
+    pub fn partial(
+        owner: AztecAddress,
+        storage_slot: Field,
+        context: &mut PrivateContext,
+        recipient: AztecAddress,
+        sender: AztecAddress,
+    ) -> PartialUintNote {
+        // Safety: We use the randomness to preserve the privacy of the note recipient by preventing brute-forcing,
+        // so a malicious sender could use non-random values to make the note less private. But they already know
+        // the full note pre-image anyway, and so the recipient already trusts them to not disclose this
+        // information. We can therefore assume that the sender will cooperate in the random value generation.
+        let randomness = unsafe { random() };
+
+        // We create a commitment to the private data, which we then use to construct the log we send to the recipient.
+        let commitment = UintPartialNotePrivateContent { owner, randomness }
+            .compute_partial_commitment(storage_slot);
+
+        // Our partial note log encoding scheme includes a field with the tag of the public completion log, and we use
+        // the commitment as the tag. This is good for multiple reasons:
+        //  - the commitment is uniquely tied to this partial note
+        //  - the commitment is already public information, so we're not revealing anything else
+        //  - we don't need to create any additional information, private or public, for the tag
+        //  - other contracts cannot impersonate us and emit logs with the same tag due to public log siloing
+        let private_log_content = PrivateUintPartialNotePrivateLogContent {
+            owner,
+            randomness,
+            public_log_tag: commitment,
+        };
+
+        // TODO: we're abusing the note encoding scheme by computing the log for a fake note type with such a note type
+        // id that the recipient will realize that these are the private fields of a partial note. Ideally we'd not rely
+        // on this crude mechanism and we'd instead compute it as a proper event log. However, given the current state
+        // of the log library it's far easier to do it this way.
+        let encrypted_log = default_aes128::note::compute_log(
+            *context,
+            private_log_content,
+            storage_slot,
+            recipient,
+            sender,
+        );
+        context.emit_private_log(encrypted_log);
```

Comment:
is there a way of not-choosing to use `default_aes128` encryption strategy here?
can the `fn partial` have as input an encryption method?

Reply:
As of today, encryption is hardcoded in PXE to be AES. We'll soon move it into aztec-nr, at which point it'd be possible to somehow configure which encryption algorithm the contracts wishes to use. But this is just a first step, so it'll take a while to get there. Keep in mind this is the very first attempt to do this this alternative way. We'd similarly want to allow for unconstrained encryption etc.